libsolidity/analysis/ControlFlowAnalyzer.cpp

```diff
@@ -110,7 +110,10 @@ void ControlFlowAnalyzer::checkUninitializedAccess(CFGNode const* _entry, CFGNod
 
 		// Propagate changes to all exits and queue them for traversal, if needed.
 		for (auto const& exit: currentNode->exits)
-			if (nodeInfos[exit].propagateFrom(nodeInfo))
+			if (
+				auto exists = valueOrNullptr(nodeInfos, exit);
+				nodeInfos[exit].propagateFrom(nodeInfo) || !exists
```

Comment:
While reviewing code before these lines, I couldn't find where `nodeInfos` is initialized. It's declared here

https://github.com/ethereum/solidity/blob/46348a306c2c9837e15502569faeae23598ad51f/libsolidity/analysis/ControlFlowAnalyzer.cpp#L69

 but I'm not sure what it contains at the point of the reviewed line.

Reply:
iiuc, `nodeInfos[exit]` references an existing key in the map or creates a new `nodeInfo` for `exit` node, adding it to the map. Why do we need the `auto exists = ...` in which case?